src/core/net/netif.cpp

```diff
@@ -42,6 +42,32 @@
 namespace ot {
 namespace Ip6 {
 
+class AddressInfo : public otIp6AddressInfo
+{
+public:
+    AddressInfo(const NetifUnicastAddress &aAddress, Instance &aInstance)
+    {
+        mAddress      = &aAddress.mAddress;
+        mPrefixLength = aAddress.mPrefixLength;
+        mScope        = aAddress.GetScope();
+        mIsAnycast    = aInstance.Get<Mle::MleRouter>().IsAnycastLocator(aAddress.GetAddress());
+        mIsRloc       = aAddress.mRloc;
```

Comment:
nit: Instead of using mutliple bit filelds (i.e. `mIsAnycast`, `mIsRloc`), what about define address types: e.g.

- RLOC
- ALOC
- MLEID
- OTHER

So, we can save more information in the two bits.

Reply:
I will remove the mIsRloc, which is not used.

Reply:
SGTM